Feat/archieve customers

mmehmetAliIzci opened the pull request:
# Changes

## Related issues

https://github.com/FrontMen/fm-hours/issues/52

## Added

New attribute for the `customer` model:

- `archived`
- `archivedDate`

## Changed

- Customer list page now has 1 more column named: `archived`.
- Customer list page now can be filtered by archived or not in combination with other filters.
- Customer detail page now has `archive` - `unarchive` button

## How to test

On customer list page, try all combinations for filtering the customers.
On customer detail page, try to archive customers and see if its reflected back.

## Screenshots
https://user-images.githubusercontent.com/7356299/124759293-c797bf80-df2f-11eb-8b6b-87458a980d97.mov

https://user-images.githubusercontent.com/7356299/124759276-c2d30b80-df2f-11eb-9d7f-ef001da8fb57.mov

YejunWu168 commented:
I would suggest to change the background color of the labels according to the archived state. So it's easier to see in a glance if a customer is archived.

YejunWu168 commented:
![Screenshot 2021-07-07 at 15 33 26](https://user-images.githubusercontent.com/37107741/124768178-bbfcc680-df38-11eb-9f9a-3193d5a6c01e.png)

mmehmetAliIzci commented on pages/customers/_customer_id.vue:
```diff
@@ -16,6 +16,7 @@
           @submit.prevent="handleSubmit"
           @change="hasUnsavedChanges = true"
         >
+          <h3 v-if="form.archived">This customer archived at {{ form.archivedDate | formatDate("dd MMMM yyyy") }}</h3>
```
Done

mmehmetAliIzci commented:
@YejunWu168 I've changed the colors of the badges,
@vladPinteaFrontmen I've fixed
- Employees shouldn't be able to add times archived customers
- Admins shouldn't be able to assign archived customers to employees
- Made the h3 => info

mmehmetAliIzci commented:
https://user-images.githubusercontent.com/7356299/124776760-fcac0e00-df3f-11eb-98d5-d063bd8d03a5.mov

vladPinteaFrontmen reviewed (approved):
updates look good!

YejunWu168 reviewed (approved):
Looks good!

mmehmetAliIzci commented:
https://user-images.githubusercontent.com/7356299/124904382-3d119780-dfe5-11b-9bde-75b7974c0a1f.mov

@vladPinteaFrontmen @YejunWu168 Could you check if this is an acceptable approach when it comes to "not copying the already archived customer's project from last week"

vladPinteaFrontmen commented:
@mmehmetAliIzci it's consistent with other alerts we have in our app, so I think it's good. The message is quite clear so it would work fine. 

What happens if the only project in the past week is archived? Do we add the default project in the next page?

mmehmetAliIzci commented:
@vladPinteaFrontmen No we don't add anything. If there is only one project which is archived, there is only alert shown and you are responsible for adding whatever you want to add
